src/main/kotlin/org/opensearch/replication/seqno/RemoteClusterRetentionLeaseHelper.kt

```diff
@@ -138,7 +196,7 @@ class RemoteClusterRetentionLeaseHelper constructor(val followerClusterName: Str
 
     public fun renewRetentionLease(leaderShardId: ShardId, seqNo: Long,
                                    followerShardId: ShardId, timeout: Long) {
-        val retentionLeaseId = retentionLeaseIdForShard(followerClusterName, followerShardId)
+        val retentionLeaseId = retentionLeaseIdForShard(followerClusterNameWithUUID, followerShardId)
```

Comment:
Why are we not considering the renew fail to old retention lease Id here?

Reply:
This function will only be called when retention lease will be added. Since we are going to always going to be adding new retention lease, we dont have to check for old retention lease here.

Reply:
Adding to that, this function is getting called by `addRetentionLease` which is in turn called only by the leader.
Once this change goes through, the follower will always send clustername:clusteruuid, this will ensure that only new retentionlease is added for any new replication that is started.